lib/datadog/core/remote/component.rb

```diff
@@ -18,6 +18,28 @@ def initialize(settings, agent_settings)
           transport_options = {}
           transport_options[:agent_settings] = agent_settings if agent_settings
 
+          transport_root = Datadog::Core::Transport::HTTP.root(**transport_options.dup)
+
+          res = transport_root.send_info
+          if res.ok?
+            if res.endpoints.include?('/v0.7/config')
+              Datadog.logger.debug { 'agent reachable and reports remote configuration endpoint' }
+            else
+              Datadog.logger.error do
+                'agent reachable but does not report remote configuration endpoint: ' \
+                  'disabling remote configuration for this process.'
+              end
```

Comment:
Since this is a user-visible message, consider perhaps suggesting that users upgrade their agent to get rid of this message?

(Same for the unreachable message below -- perhaps tell them that something seems wrong and they'll probably not be able to report data?)

Reply:
Great point @ivoanjo 

Since we are targeting this change beta release. We can improve the error message for the official release 😄